StreamChatClient_v3/Utils/Codable+Extensions.swift

```diff
@@ -144,3 +144,17 @@ struct AnyEncodable: Encodable {
         try encodable.encode(to: encoder)
     }
 }
+
+// MARK: - Safe Helpers
+
+extension Encodable {
+    func encodeSafely(to encoder: Encoder, logMessage: String? = nil) {
+        do {
+            try encode(to: encoder)
+        } catch {
+            if let logMessage = logMessage {
+                log.error("\(logMessage): \(encoder)")
+            }
+        }
+    }
+}
```

Comment:
This shouldn't be needed. In v2 the extra data is optional, in v3 it's mandatory. You should be able to just encode the extra data and that's it.